### PR TITLE
west.yml: Update MCUboot to fix serial recovery issues

### PR DESCRIPTION
**Pull request by MarekPieta**
Change updates MCUboot to remove legacy GPIO configuration. The legacy configuration became problematic, because GPIO DTS nodes no longer support labels that were used to identify nodes in MCUboot.

The GPIO should be configured by board's DTS, which is simpler.

ref. NCSDK-16995

### CODE REVIEW

**Review by nvlsianpu — APPROVED** [review]

File: west.yml [hunk 1]
```diff
@@ -103,7 +103,7 @@ manifest:
       revision: 47399f055431513b5dfcd9d1fd1ec5dc5c96a252
     - name: mcuboot
       repo-path: sdk-mcuboot
-      revision: v1.9.99-ncs2-rc2
+      revision: pull/214/head
```
Comment (rlubos) [inline west.yml:106]: Please update SHA: `bf00840a090f396ec1554968e19fa0e02c077d38`

Comment (MarekPieta) [reply west.yml]: Done

**Review by rlubos — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]
lgtm

**Comment by rlubos** [comment]
CI needed a rerun, but passed after that.

**Pull request merged**